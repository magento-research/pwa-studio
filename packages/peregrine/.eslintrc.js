const config = {
    parser: 'babel-eslint',
    extends: ['@magento'],
    rules: {
        "no-undef": "off"
    }
};

module.exports = config;
