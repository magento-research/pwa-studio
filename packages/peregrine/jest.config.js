module.exports = {
    displayName: 'Peregrine',
    setupFiles: ['<rootDir>/scripts/shim.js', '<rootDir>/scripts/fetch-mock.js']
};
