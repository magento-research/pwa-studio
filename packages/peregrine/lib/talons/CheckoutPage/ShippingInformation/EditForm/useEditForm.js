import { useCallback } from 'react';
import { useMutation } from '@apollo/react-hooks';

import { useCartContext } from '../../../../context/cart';

export const useEditForm = props => {
    const {
        mutations: { setShippingInformationMutation },
        shippingData
    } = props;

    const [{ cartId }] = useCartContext();
    const [setShippingInformation, { called, loading }] = useMutation(
        setShippingInformationMutation
    );

    const { country, region } = shippingData;
    const { code: countryCode } = country;
    const { code: regionCode } = region;

    const initialValues = {
        ...shippingData,
        country: countryCode,
        region: regionCode
    };

    const handleSubmit = useCallback(
        formValues => {
            const { country, email, ...address } = formValues;
            setShippingInformation({
                variables: {
                    cartId,
                    email,
                    address: {
                        ...address,
                        country_code: country
                    }
                }
            });
        },
        [cartId, setShippingInformation]
    );

    return { handleSubmit, initialValues, isSaving: called && loading };
};
