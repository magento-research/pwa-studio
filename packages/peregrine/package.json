{
  "name": "@magento/peregrine",
  "version": "3.0.0",
  "publishConfig": {
    "access": "public"
  },
  "description": "The core runtime of Magento PWA",
  "main": "lib/index.js",
  "scripts": {
    "build": " ",
    "build:ci": " ",
    "build:cjs": " ",
    "build:dev": " ",
    "clean": " ",
    "storybook": "start-storybook -p 9001 -c .storybook",
    "storybook:build": "build-storybook -c .storybook -o storybook-dist",
    "watch": " "
  },
  "repository": "github:magento-research/pwa-studio",
  "author": "Magento Commerce",
  "license": "(OSL-3.0 OR AFL-3.0)",
  "bugs": {
    "url": "https://github.com/magento-research/pwa-studio/issues"
  },
  "homepage": "https://github.com/magento-research/pwa-studio/tree/master/packages/peregrine#readme",
  "devDependencies": {
    "@storybook/react": "~4.1.13",
    "intl": "~1.2.5",
    "react": "~16.8.3",
    "react-apollo": "~2.5.3",
    "react-dom": "~16.8.3",
    "react-router-dom": "~5.0.0",
    "react-test-renderer": "~16.8.6",
    "rimraf": "~2.6.3"
  },
  "peerDependencies": {
    "@babel/runtime": "^7.4.2",
    "react": "^16.8.0",
    "react-apollo": "^2.5.3",
    "react-dom": "^16.8.0",
    "react-router-dom": "~5.0.0"
  },
  "engines": {
    "node": "8.10.0 || >=10.7.0",
    "yarn": ">=1.12.0"
  },
  "module": "lib/index.js",
  "jsnext:main": "lib/index.js",
  "es2015": "lib/index.js",
  "sideEffects": false
}
