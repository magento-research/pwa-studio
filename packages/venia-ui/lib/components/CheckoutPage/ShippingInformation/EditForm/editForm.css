.root {
    display: grid;
    gap: 0.5rem 1.5rem;
    grid-template-columns: 1fr 1fr;
    width: 100%;
}

.field {
    grid-column: 1 / span 2;
}

.firstname {
    grid-column: 1;
}

.lastname {
    grid-column: 2;
}

.submit {
    composes: root_normalPriority from '../../../Button/button.css';
    font-size: 0.875rem;
    font-weight: 600;
    grid-column: 1 / span 2;
    justify-self: center;
    margin-top: 1rem;
}

@media (max-width: 960px) {
    .firstname {
        grid-column: 1 / span 2;
    }

    .lastname {
        grid-column: 1 / span 2;
    }
}
