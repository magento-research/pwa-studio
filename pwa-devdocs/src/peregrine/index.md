---
title: Peregrine
---

The Peregrine project is a collection of UI Components for Magento PWA projects.
Use, extend, or remix these components to create a unique Magento PWA storefront.

Visit the [`Peregrine`] repository to contribute to this project.

[`Peregrine`]: https://github.com/magento-research/pwa-studio/tree/master/packages/peregrine
