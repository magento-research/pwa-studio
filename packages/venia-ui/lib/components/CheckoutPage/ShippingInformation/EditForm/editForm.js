import React from 'react';
import { Form } from 'informed';
import { useEditForm } from '@magento/peregrine/lib/talons/CheckoutPage/ShippingInformation/EditForm/useEditForm';

import { mergeClasses } from '../../../../classify';
import { isRequired } from '../../../../util/formValidators';
import Button from '../../../Button';
import Country from '../../../Country';
import Field, { Message } from '../../../Field';
import Region from '../../../Region';
import TextInput from '../../../TextInput';
import defaultClasses from './editForm.css';
import EditFormOperations from './editForm.gql';

const EditForm = props => {
    const { afterSubmit, classes: propClasses, shippingData } = props;
    const talonProps = useEditForm({
        afterSubmit,
        ...EditFormOperations,
        shippingData
    });
    const { handleSubmit, initialValues, isSaving, isUpdate } = talonProps;
    const classes = mergeClasses(defaultClasses, propClasses);

    return (
        <Form
            className={classes.root}
            initialValues={initialValues}
            onSubmit={handleSubmit}
        >
            <Field classes={{ root: classes.field }} id="email" label="Email">
                <TextInput field="email" validate={isRequired} />
                <Message fieldState={{}}>
                    Set a password at the end of guest checkout to create an
                    account in one easy step.
                </Message>
            </Field>
            <Field
                classes={{ root: classes.firstname }}
                id="firstname"
                label="First Name"
            >
                <TextInput field="firstname" validate={isRequired} />
            </Field>
            <Field
                classes={{ root: classes.lastname }}
                id="lastname"
                label="Last Name"
            >
                <TextInput field="lastname" validate={isRequired} />
            </Field>
            <Country classes={{ root: classes.field }} validate={isRequired} />
            <Field
                classes={{ root: classes.field }}
                id="street0"
                label="Street Address"
            >
                <TextInput field="street[0]" validate={isRequired} />
            </Field>
            <Field
                classes={{ root: classes.field }}
                id="street1"
                label="Street Address 2"
            >
                <TextInput field="street[1]" />
            </Field>
            <Field classes={{ root: classes.field }} id="city" label="City">
                <TextInput field="city" validate={isRequired} />
            </Field>
            <Region classes={{ root: classes.field }} validate={isRequired} />
            <Field
                classes={{ root: classes.field }}
                id="postcode"
                label="ZIP / Postal Code"
            >
                <TextInput field="postcode" validate={isRequired} />
            </Field>
            <Field
                classes={{ root: classes.field }}
                id="telephone"
                label="Phone Number"
            >
                <TextInput field="telephone" validate={isRequired} />
            </Field>
            <Button
                classes={{ root_normalPriority: classes.submit }}
                disabled={isSaving}
                priority="normal"
                type="submit"
            >
                {isUpdate ? 'Update' : 'Continue to Shipping Method'}
            </Button>
        </Form>
    );
};

EditForm.defaultProps = {
    shippingData: {
        country: {
            code: 'US'
        },
        region: {
            code: ''
        }
    }
};

export default EditForm;
