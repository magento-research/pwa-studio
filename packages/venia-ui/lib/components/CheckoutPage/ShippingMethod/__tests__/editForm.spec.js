import React from 'react';
import { createTestInstance } from '@magento/peregrine';

import EditForm from '../editForm';

jest.mock('../../../../classify');
jest.mock('../shippingRadios', () => 'Shipping Radios Component');

test('it renders correctly', () => {
    // Act.
    const instance = createTestInstance(
        <EditForm
            handleCancel={jest.fn()}
            handleSubmit={jest.fn()}
            isLoading={false}
            mode={'initial'}
            pageIsUpdating={false}
            selectedShippingMethod={'flatrate|flatrate'}
            shippingMethods={[
                {
                    amount: {
                        currency: 'USD',
                        value: 99
                    },
                    available: true,
                    carrier_code: 'flatrate',
                    carrier_title: 'Flat Rate',
                    method_code: 'flatrate',
                    method_title: 'Flat Rate',
                    serializedValue: 'flatrate|flatrate'
                }
            ]}
        />
    );

    // Assert.
    expect(instance.toJSON()).toMatchSnapshot();
});

test('it renders correctly in update mode', () => {
    // Act.
    const instance = createTestInstance(
        <EditForm
            handleCancel={jest.fn()}
            handleSubmit={jest.fn()}
            isLoading={false}
            mode={'update'}
            pageIsUpdating={false}
            selectedShippingMethod={'flatrate|flatrate'}
            shippingMethods={[
                {
                    amount: {
                        currency: 'USD',
                        value: 99
                    },
                    available: true,
                    carrier_code: 'flatrate',
                    carrier_title: 'Flat Rate',
                    method_code: 'flatrate',
                    method_title: 'Flat Rate',
                    serializedValue: 'flatrate|flatrate'
                }
            ]}
        />
    );

    // Assert.
    expect(instance.toJSON()).toMatchSnapshot();
});
