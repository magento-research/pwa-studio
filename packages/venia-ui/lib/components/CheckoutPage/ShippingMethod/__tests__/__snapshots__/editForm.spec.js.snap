// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders correctly 1`] = `
<form
  className="root"
  onKeyDown={[Function]}
  onReset={[Function]}
  onSubmit={[Function]}
>
  <Shipping Radios Component
    isLoading={false}
    selectedShippingMethod="flatrate|flatrate"
    shippingMethods={
      Array [
        Object {
          "amount": Object {
            "currency": "USD",
            "value": 99,
          },
          "available": true,
          "carrier_code": "flatrate",
          "carrier_title": "Flat Rate",
          "method_code": "flatrate",
          "method_title": "Flat Rate",
          "serializedValue": "flatrate|flatrate",
        },
      ]
    }
  />
  <div
    className="initialButtonContainer"
  >
    <button
      className="root_normalPriority"
      disabled={false}
      type="submit"
    >
      <span
        className="content"
      >
        Continue to Payment Information
      </span>
    </button>
  </div>
</form>
`;

exports[`it renders correctly in update mode 1`] = `
<form
  className="root"
  onKeyDown={[Function]}
  onReset={[Function]}
  onSubmit={[Function]}
>
  <Shipping Radios Component
    isLoading={false}
    selectedShippingMethod="flatrate|flatrate"
    shippingMethods={
      Array [
        Object {
          "amount": Object {
            "currency": "USD",
            "value": 99,
          },
          "available": true,
          "carrier_code": "flatrate",
          "carrier_title": "Flat Rate",
          "method_code": "flatrate",
          "method_title": "Flat Rate",
          "serializedValue": "flatrate|flatrate",
        },
      ]
    }
  />
  <div
    className="updateButtonContainer"
  >
    <button
      className="root_normalPriority"
      onClick={[MockFunction]}
      type="button"
    >
      <span
        className="content"
      >
        Cancel
      </span>
    </button>
    <button
      className="root_highPriority"
      disabled={false}
      type="submit"
    >
      <span
        className="content"
      >
        Update
      </span>
    </button>
  </div>
</form>
`;
